src/processTargets.ts

```diff
@@ -207,20 +207,24 @@ function transformSelection(
           });
         }
       } else if (transformation.pieceType === "character") {
-        const characters = token.split("");
-        for (let index = 0; index < characters.length; index++) {
-          pieces.push({ start: index, end: index + 1 });
-        }
+        pieces = range(token.length).map((index) => ({
+          start: index,
+          end: index + 1,
+        }));
       }
 
       // NB: We use the modulo here to handle negative offsets
-      let endIndex =
+      const endIndex =
         transformation.endIndex == null
           ? pieces.length
-          : (transformation.endIndex + pieces.length) % pieces.length;
-      endIndex = endIndex === 0 ? pieces.length : endIndex;
+          : transformation.endIndex <= 0
+          ? transformation.endIndex + pieces.length
+          : transformation.endIndex;
```

Comment:
I think this is a bit clearer than what I had originally; I think the modulo thing was too clever, as evidenced by the fact that it was broken 😅